### PR TITLE
Use number_of_shards instead of shard_count

### PR DESCRIPTION
**Pull request by dliappis**
Since standard rally-tracks (https://github.com/elastic/rally-tracks)
use `number_of_shards` as the track parameter for change the number of
primary shards, this commit applies this convention to this track as
well.

### CODE REVIEW

**Comment by dliappis** [comment]
I added >1 reviewers for general awareness.

File: README.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ The table below shows the track parameters that can be adjusted along with defau
 | Parameter | Explanation | Type | Default Value |
 | --------- | ----------- | ---- | ------------- |
 | `number_of_replicas` | Number of index replicas | `int` | `0` |
-| `shard_count` | Number of primary shards | `int` | `2` |
+| `number_of_shards` | Number of primary shards | `int` | `2` |
```
Comment (danielmitterdorfer) [inline README.md]: This is obsolete here in the sense that this challenge does not set a different number of shards (the default in the index template is two shards already). I think we can safely remove the whole line here and instead document this as a track parameter available for all challenges? This applies similarly to `number_of_replicas`. However, we can merge this PR as is and do the cleanup in a followup PR.

File: README.md [hunk 2]
```diff
@@ -73,7 +73,7 @@ The table below shows the track parameters that can be adjusted along with defau
...
```
Comment (danielmitterdorfer) [inline README.md]: This is obsolete here in the sense that this challenge does not set a different number of shards (the default in the index template is two shards already). I think we can safely remove the whole line here and instead document this as a track parameter available for all challenges? This applies similarly to `number_of_replicas`. However, we can merge this PR as is and do the cleanup in a followup PR.

File: README.md [hunk 3]
```diff
@@ -94,7 +94,7 @@ The table below shows the track parameters that can be adjusted along with defau
...
```
Comment (danielmitterdorfer) [inline README.md]: This is obsolete here in the sense that this challenge does not set a different number of shards (the default in the index template is two shards already). I think we can safely remove the whole line here and instead document this as a track parameter available for all challenges? This applies similarly to `number_of_replicas`. However, we can merge this PR as is and do the cleanup in a followup PR.

File: README.md [hunk 4]
```diff
@@ -111,7 +111,7 @@ The table below shows the track parameters that can be adjusted along with defau
...
```
Comment (danielmitterdorfer) [inline README.md]: This is obsolete here in the sense that this challenge does not set a different number of shards (the default in the index template is two shards already). I think we can safely remove the whole line here and instead document this as a track parameter available for all challenges? This applies similarly to `number_of_replicas`. However, we can merge this PR as is and do the cleanup in a followup PR.

File: eventdata/challenges/large-shard-sizing.json [hunk 5]
```diff
@@ -26,7 +26,7 @@
       "operation": {
         "operation-type": "create-index-template",
         "settings": {
-          "index.number_of_shards": {{ shard_count | default(1) }}
+          "index.number_of_shards": {{ number_of_shards | default(1) }}
```
Comment (danielmitterdorfer) [inline eventdata/challenges/large-shard-sizing.json]: The change from the default is not documented. However, I'm ok if we do this in a follow-up PR.

File: eventdata/challenges/document_id_benchmark.json [hunk 6]
```diff
@@ -26,7 +26,7 @@
         "operation-type": "create-index-template",
         "settings": {
             "index.translog.retention.size": "10mb",
-            "index.number_of_shards": {{ shard_count | default(1) }}
+            "index.number_of_shards": {{ number_of_shards | default(1) }}
```
Comment (danielmitterdorfer) [inline eventdata/challenges/document_id_benchmark.json]: The change from the default is not documented. However, I'm ok if we do this in a follow-up PR.

**Comment by dliappis** [comment]
NOTE: Once this gets merged, https://github.com/elastic/rally/pull/876 needs to be merged immediately to ensure IT tests don't fail.

**Pull request merged**